proto/scheduler.proto

```diff
@@ -24,13 +24,21 @@ message LeaseTaskRequest {
   // Mutually exclusive with `release`.
   bool finalize = 2;
 
+  // DEPRECATED
   // Indicates that the lease should be released without finalizing (deleting)
   // the task.
   // Mutually exclusive with `finalize`.
   bool release = 3;
 
   // ID of the executor making the request.
   string executor_id = 4;
+
+  // Indicates that the leased task could not be run to completition and should
+  // be re-enqueued to be retried.
+  bool re_enqueue = 5;
+  // Optional description of why the task needs to be re-enqueued (may be
+  // visible to end user).
+  string re_enqueue_reason = 6;
```

Comment:
Could we use a proto.Status here so we get the code too?

Reply:
Done.